src/ucs/sys/topo.c

```diff
@@ -96,11 +97,86 @@ ucs_status_t ucs_topo_find_device_by_bus_id(const ucs_sys_bus_id_t *bus_id,
     return UCS_OK;
 }
 
+static void ucs_topo_get_path_with_bus(unsigned bus, char *path)
+{
+    char prefix[] = "/sys/class/pci_bus";
+
+    if (bus < 16) {
+        sprintf(path, "%s/0000:0%x", prefix, bus);
+    } else {
+        sprintf(path, "%s/0000:%x", prefix, bus);
+    }
+}
 
 ucs_status_t ucs_topo_get_distance(ucs_sys_device_t device1,
                                    ucs_sys_device_t device2,
                                    ucs_sys_dev_distance_t *distance)
 {
+    int score = 0;
+    int depth = 0;
+    int same  = 1;
+	int i;
+    int comp_len;
+    char path1[PATH_MAX], path2[PATH_MAX];
+    char resolved_path1[PATH_MAX], resolved_path2[PATH_MAX];
+    unsigned bus1, bus2;
+    int rp_len1, rp_len2;
+
+    /* TODO: Use generic distance queury after 1.9 release but use 1.6 WAR now
+     */
+
+    bus1 = (ucs_topo_ctx.sys_dev_to_bus_lookup.bus_arr[device1]).bus;
+    bus2 = (ucs_topo_ctx.sys_dev_to_bus_lookup.bus_arr[device2]).bus;
+
+    ucs_topo_get_path_with_bus(bus1, path1);
+    ucs_topo_get_path_with_bus(bus2, path2);
+
+    if (NULL == realpath(path1, resolved_path1)) {
+        return UCS_ERR_IO_ERROR;
+    }
+
+    if (NULL == realpath(path2, resolved_path2)) {
+        return UCS_ERR_IO_ERROR;
+    }
+
+    ucs_debug("path1 = %s, resolved_path = %s", path1, resolved_path1);
+    ucs_debug("path2 = %s, resolved_path = %s", path2, resolved_path2);
+
+    rp_len1  = strlen(resolved_path1);
+    rp_len2  = strlen(resolved_path2);
+    comp_len = (rp_len1 <= rp_len2) ? rp_len1 : rp_len2;
+
+    for (i = 0; i < comp_len; i++) {
+        if (resolved_path1[i] != resolved_path2[i]) same = 0;
+        if (resolved_path1[i] == '/') {
+            depth++;
+            if (same) {
+                score++;
+            }
+        }
+    }
+
+    if (3 == score) {
```

Comment:
what is 3 , 4 here? can we define macro?

Reply:
can you use switch() and use single ucs_debug before return from the func

Reply:
@bureddy Simplified this logic to return relative scoring instead. Can you check if it makes sense?